pkg/compose/build.go

```diff
@@ -352,3 +358,26 @@ func addSecretsConfig(project *types.Project, service types.ServiceConfig) (sess
 	}
 	return secretsprovider.NewSecretProvider(store), nil
 }
+
+func addPlatforms(project *types.Project, service types.ServiceConfig) ([]specs.Platform, error) {
+	var plats []specs.Platform
+	if platform, ok := project.Environment["DOCKER_DEFAULT_PLATFORM"]; ok {
```

Comment:
Should we respect this variable on `up` / `run` as well?

```shell
$ DOCKER_DEFAULT_PLATFORM="linux/amd64" docker compose up --build --quiet-pull --force-recreate hello

[+] Running 1/1
 ⠿ Container platform-hello-1  Recreated                                                                                                               0.1s
Attaching to platform-hello-1
platform-hello-1  | hello linux arm64
platform-hello-1 exited with code 0
```

Reply:
yes! thanks @milas